code/__DEFINES/dcs/signals.dm

```diff
@@ -251,13 +251,15 @@
 
 // /obj/item/weapon/gun signals
 #define COMSIG_GUN_FIRE "gun_fire"
-	#define COMPONENT_GUN_FIRED 1
-#define COMSIG_GUN_AUTOFIRE "gun_autofire"
-#define COMSIG_GUN_CLICKEMPTY "gun_clickempty"
-#define COMSIG_GUN_FIREMODE_TOGGLE "gun_firemode_toggle"		//from /obj/item/weapon/gun/verb/toggle_firemode()
-#define COMSIG_GUN_FIREDELAY_MODIFIED "gun_firedelay_modified"
-#define COMSIG_GUN_BURSTDELAY_MODIFIED "gun_burstdelay_modified"
-#define COMSIG_GUN_BURSTAMOUNT_MODIFIED "gun_burstamount_modified"
+#define COMSIG_GUN_STOP_FIRE "gun_stop_fire"
+#define COMSIG_GUN_MUST_FIRE "gun_must_fire"
+	#define GUN_HAS_FIRED	(1<<0)
+#define COMSIG_GUN_FIRE_MODE_TOGGLE "gun_firemode_toggle"		//from /obj/item/weapon/gun/verb/toggle_firemode()
+#define COMSIG_GUN_AUTOFIREDELAY_MODIFIED "gun_firedelay_modified"
+#define COMSIG_GUN_BURST_SHOTS_TO_FIRE_MODIFIED "gun_burstamount_modified"
+#define COMSIG_GUN_BURST_SHOT_DELAY_MODIFIED "gun_burstdelay_modified"
+#define COMSIG_GUN_FIRE_RESET "gun_fire_reset"
+#define COMSIG_GUN_AUTO_FIRE_COMP_RESET "gun_auto_fire_reset"
```

Comment:
some of these signals such as 
COMSIG_GUN_FIRE_RESET
are basically just proccall replacements please make it more generic